### PR TITLE
Strings dot-syntax-swift3 fix

### LINKED ISSUE #241: Fix naming issues in strings dot-syntax-swift3 template
Currently using the `-t dot-syntax-swift3` option leads to the following code in usage side:
```swift
L10n.Bananas.owner(p0: 2, p1: "John")
```

Instead it should be:
```swift
L10n.Bananas.owner(2, "John")
```

See https://github.com/AliSoftware/SwiftGen/pull/168#discussion_r92412861 on how this issue was introduced and where it lies. We should wait for #237 to be merged before fixing this issue though as suggested by @AliSoftware [here](https://github.com/AliSoftware/SwiftGen/pull/168#discussion_r92413697).

### PR DESCRIPTION
**Pull request by djbe**
Fixes #241, is also an example on how to get rid of the `p0`, `p1` variables.

### CODE REVIEW

File: templates/strings-dot-syntax-swift3.stencil [hunk 1]
```diff
@@ -10,12 +10,14 @@ import Foundation
 // swiftlint:disable nesting
 // swiftlint:disable variable_name
 // swiftlint:disable valid_docs
+{% macro parametersBlock params %}{% for type in params.types %}_ p{{forloop.counter}}: {{type}}{% if not forloop.last %}, {% endif %}{% endfor %}{% endmacro %}
+{% macro argumentsBlock params %}{% for type in params.types %}p{{forloop.counter}}{% if not forloop.last %}, {% endif %}{% endfor %}{% endmacro %}
```
Comment (AliSoftware) [inline templates/strings-dot-syntax-swift3.stencil]: Why not use Stencil's `|join(", ")` filter here instead of a `for` loop?
Would avoid the use of `if not forloop.last`…

File: templates/strings-dot-syntax-swift3.stencil [hunk 2]
```diff
@@ -10,12 +10,14 @@ import Foundation
 // swiftlint:disable nesting
 // swiftlint:disable variable_name
 // swiftlint:disable valid_docs
+{% macro parametersBlock params %}{% for type in params.types %}_ p{{forloop.counter}}: {{type}}{% if not forloop.last %}, {% endif %}{% endfor %}{% endmacro %}
```
Comment (AliSoftware) [inline templates/strings-dot-syntax-swift3.stencil]: Wondering what it would like (i.e. would it be more readable or less readable?) if you used a `MapNode` here instead?

File: templates/strings-dot-syntax-swift3.stencil (hunk 1 above)
Comment (djbe) [inline templates/strings-dot-syntax-swift3.stencil]: Because I don't have an array at that point? I'm trying to __not__ use the names/namedtypes variables.

File: templates/strings-dot-syntax-swift3.stencil (hunk 2 above)
Comment (AliSoftware) [inline templates/strings-dot-syntax-swift3.stencil]: Ah, right. Might be worth nitpicking it in a separate PR btw (that we could merge either before or after the Great Split and the next major version, depending on our agenda)

File: templates/strings-dot-syntax-swift3.stencil (hunk 1 above)
Comment (AliSoftware) [inline templates/strings-dot-syntax-swift3.stencil]: Ah, right, sorry I talked too fast, was thinking about `params.types|join(", ")` but that's not what you want, you don't want `Int, Float, String` but `p0, p1, p2` here, right. nvm

**Comment by AliSoftware** [comment]
I've edited #244 to add `names` and `typednames` to your list btw

**Comment by djbe** [comment]
Cool! Actually that whole `params` array of sub-items can just become the contents of `params.types`. The rest is not needed anymore.

**Comment by AliSoftware** [comment]
Haha I see you've won the race and managed to validate your edit to add that new consideration… before I did validate my own edits and got rejected ^^

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,7 +6,10 @@
 
 ### Bug Fixes
 
-_None_
+* Strings: fix issue with `dot-syntax-swift3` where function definitions were
+  not swift3 guidelines compliant.
```
Comment (AliSoftware) [inline CHANGELOG.md]: `s/swift3/Swift 3/` ?
